flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/optimize/BatchCommonSubGraphBasedOptimizer.scala

```diff
@@ -80,12 +81,17 @@ class BatchCommonSubGraphBasedOptimizer(planner: BatchPlanner)
       .getOrElse(FlinkBatchProgram.buildProgram(config.getConfiguration))
     Preconditions.checkNotNull(programs)
 
+    val context = relNode.getCluster.getPlanner.getContext.unwrap(classOf[FlinkContext])
+
```

Comment:
Maybe we can inline this code with line 92 ~ 94

Reply:
Not inline because:
I think it is better to get all fields from this context instead of getting from planner.
`BatchOptimizeContext` should extends previous context.
Not do it because 1.10.